docs/developer/architecture/centralized-cli.rst

```diff
@@ -0,0 +1,172 @@
+|            |                  | roles, manifest-template-path           |
++------------+------------------+-----------------------------------------+
+| the manifest-template-path is the path to the manifest describing the   |
+| resource. Everything that is in the command line, override the manifest |
+| content. This apply to Node and Volumes.                                |
++------------+------------------+-----------------------------------------+
+| node       | edit             | name, hostname (to change node hostname)|
++------------+------------------+-----------------------------------------+
+| node       | deploy           | <list of nodes>, dry-run                |
++------------+------------------+-----------------------------------------+
+| node       | delete, drain,   | <list of nodes>                         |
+|            | taint, label     |                                         |
++------------+------------------+-----------------------------------------+
+| volume     | add              | name, nodeName, storageClassName,       |
+|            |                  | devicePath, manifest-template-path,     |
+|            |                  | labels, type                            |
++------------+------------------+-----------------------------------------+
+| volume     | delete, labels   | name                                    |
++------------+------------------+-----------------------------------------+
+
+
+Cluster Administration
+^^^^^^^^^^^^^^^^^^^^^^
+
+**tool: mk8sctl**
+
++------------+------------+-----------------------------------------------+
+| Resource   | action     | parameters                                    |
++============+============+===============================================+
+| bootstrap  | deploy     |                                               |
++------------+------------+-----------------------------------------------+
+| cluster    | import-iso | path_to_iso                                   |
++------------+------------+-----------------------------------------------+
+| cluster    | upgrade    | dest-version, dry-run                         |
++------------+------------+-----------------------------------------------+
+| cluster    | downgrade  | dest-version, dry-run                         |
++------------+------------+-----------------------------------------------+
+| etcd       | health     |                                               |
++------------+------------+-----------------------------------------------+
+| bootstrap  | backup     |                                               |
++------------+------------+-----------------------------------------------+
+| bootstrap  | restore    | backup-file                                   |
++------------+------------+-----------------------------------------------+
+| cluster    | edit-conf  | conf-param=conf-value                         |
+|            |            | ex: ip-in-ip=true                             |
```

Comment:
I'm pretty sure you'll run into trouble with this.

`BootstrapConfiguration` is a structured object, which here you're flattening into... what? Which subactions would you support?...

What you'll want instead is a config check (is the YAML valid?), and a (re)deploy-with-new-config.

Reply:
will remove this one